pkg/controller/catalog/storage/mem_storage_service_class.go

```diff
@@ -0,0 +1,47 @@
+package storage
+
+import (
+	"errors"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+var (
+	errNoSuchServiceClass = errors.New("no such service class")
+)
+
+type memStorageServiceClass struct {
+	// This gets fetched when a SB is created (or possibly later when refetched).
+	// It's static for now to keep compatibility, seems like this could be more dynamic.
+	classes map[string]*servicecatalog.ServiceClass
+}
+
+func newMemStorageServiceClass() *memStorageServiceClass {
+	return &memStorageServiceClass{classes: make(map[string]*servicecatalog.ServiceClass)}
+}
+
+func (m *memStorageServiceClass) List() ([]*servicecatalog.ServiceClass, error) {
+	ret := make([]*servicecatalog.ServiceClass, len(m.classes))
+	i := 0
+	for _, val := range m.classes {
```

Comment:
Per https://blog.golang.org/go-maps-in-action:

> Maps are not safe for concurrent use: it's not defined what happens when you read and write to them simultaneously.

So yes. I believe you are right. And I think there might be other spots too where that applies.

@arschles if you concur, I'll fix this with a `RWMutex`.